File: 1893-check-if-all-the-integers-in-a-range-are-covered/1893-check-if-all-the-integers-in-a-range-are-covered.py
class Solution:
    def isCovered(self, ranges: list[list[int]], left: int, right: int) -> bool:
        diff = [0] * 52

        for start, end in ranges:
            diff[start] += 1
            diff[end + 1] -= 1

        active = 0
        for i in range(1, 51):
            active += diff[i]
            if left <= i <= right and active == 0:
                return False

        return True

File: 1893-check-if-all-the-integers-in-a-range-are-covered/1893-check-if-all-the-integers-in-a-range-are-covered.java
class Solution {
    public boolean isCovered(int[][] ranges, int left, int right) {
        int[] line = new int[52];
        int n = ranges.length;

        for (int[] range : ranges) {
            line[range[0]]++;
            line[range[1] + 1]--;
        }

        int overlap = 0;
        for (int i = 1; i<=right; i++) {
            overlap += line[i];

            if (i >= left && overlap == 0) {
                return false;
            }
        }

        return true;
    }
}